docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+## Do keep side effects out of your index.js
+
+When not able to keep code with side effects out of your entry point, be sure to contain such commands within a method (e.g. `run()`, `init()`, etc.) to prevent those side effects from occurring at run time.
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure your package as you do. To prevent dependency pointing from causing issue across different environments be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers, always
+
+When not pointing directly the entry point files referenced by "bar" import specifiers, be sure to reference files (both in your package and your dependencies) with their complete file extension; e.g. `import { ifDefined } from 'lit-html/directives/if-defined.js';`
+
+## Optional: use `type: "module"`
+
+If scripting in your packages is resilient to the realities made true in Node@12+ when using `type: "module"` in your `package.json`, you should use it. If all of the JS in your package is es modules, then this is a cheap win for your package and as it can support tooling the references your package in leveraging it more efficiently, however excluding it is a simple way in the near term to not require this of all of the JS your write.
+
+## Optional: include typings
+
+`*.d.ts` files are an important bridge between your code and users that choose to develop in TypeScript. If you are already developing in TypeScript, then generating these is as simple as adding the appropriate property to your `tsconfig.json`. If you are developing in JS, it can still be useful to include typings; learn more about [generating type definitions from javascript](https://dev.to/open-wc/generating-typescript-definition-files-from-javascript-5bp2).
+
+## Optional: include source maps
+
+If you choose work in TypeScript, or to use syntax that is newer than the standards discussed above, it can be important to include source maps and the source files (TypeScript or otherwise) that your published code originates from to make understanding that code, as well as possibly reporting/fixing issues in that code, easier for your users.
+
+## Do not bundle
+
+If you bundle, and then another dependency bundles and both dependencies have bundles the same transitive dependencies it will become next to impossible for later tooling to deduplicate that code. Be sure NOT to bundle code in your published packages so that your users will have the easiest time possible optimizing their application for the specific use cases they target.
```

Comment:
```suggestion
If you bundle, and then another dependency bundles and both dependencies bundle the same transitive dependencies it will become next to impossible for later tooling to deduplicate that code. Be sure NOT to bundle code in your published packages so that your users will have the easiest time possible optimizing their application for the specific use cases they target.
```

should we add an extra note or TL;DR here that "bundling is an app-level concern"?

Reply:
This is a good idea. Would it make sense to include "bundling is an app-level concern" before (or right after) the list of do's/don't's?

Reply:
I think thats a good idea yeah.

I also think its not a bad idea to have that line return a couple of times (e.g. in dont minify, dont bundle), it really does drive the whole point home.